chemprop/cli/common.py

```diff
@@ -139,6 +146,19 @@ def add_common_args(parser: ArgumentParser) -> ArgumentParser:
 
 
 def process_common_args(args: Namespace) -> Namespace:
+    if args.config_path is None:
+        return args
+
+    with open(args.config_path, "r") as f:
+        config_args = json.load(f)
+
+    for key, value in config_args.items():
+        setattr(args, key, value)
```

Comment:
note that this is the opposite behavior of standard programs. That is, if an argument is specified both on the command line and in the config file, the command line should take precedence. This is desirable because if you're given a config file and want to make just one small change, you can just specify it on the command line rather than editing the actual file (it could be read-only.) This implementation has the config file take precedence, which is not desirable.

Reply:
@davidegraff Thanks for the suggestion! This is a little more complicated than I expect. It turns out to be non-trivial to tell if the arguments are supplied by the user or just the default values. I have implemented a more generalizable way to handle this in `main.py`.